### PR TITLE
Feature/product brand

### PR DESCRIPTION
**Pull request by lbebber**
#### What is the purpose of this pull request?
Adds ProductBrand component, along with `product-brand` block.

Depends on another PR on `store-graphql` (which hasnt been created yet)

With logo: https://lbebber--alssports.myvtex.com/pat84217/p?skuId=412537
Without logo (falls back to text): https://lbebber--alssports.myvtex.com/northf-jacket-ventrix-wmns/p

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/ProductBrand/urlHelpers.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+export const DEFAULT_WIDTH = 'auto'
```
Comment (lbebber) [inline react/components/ProductBrand/urlHelpers.js]: This file has been copied from somewhere else (and from the looks of it, has been replicated several times already)

This is supposed to be temporary—a smart `VTEXImage` component should be created in the near future that handles this. Just needs its interface to be ironed out.

File: react/components/ProductBrand/index.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+import React, { useContext } from 'react'
+import { ProductContext } from 'vtex.product-context'
+import { generateBlockClass } from '@vtex/css-handles'
+import { Query } from 'react-apollo'
+import gql from 'graphql-tag'
+import { changeImageUrlSize } from './urlHelpers'
+
+import styles from './productBrand.css'
+
+const BRAND_LOGO_QUERY = gql`
```
Comment (klzns) [inline react/components/ProductBrand/index.js]: Make it external file so it's compiled at build time.

File: react/components/ProductBrand/index.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+   * server handles resizing. */
+  // width,
+  height = 100,
+  excludeBrands,
+  blockClass,
+}) => {
+  const productContext = useContext(ProductContext)
+
+  const {
+    product: { brand: brandName, brandId },
```
Comment (klzns) [inline react/components/ProductBrand/index.js]: How it's dealing when product is loading or error?

File: react/components/ProductBrand/index.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  } = productContext
+
+  /** Certain brands (e.g. placeholder brands) can be filtered out via theme config */
+  if (shouldExcludeBrand(brandName, brandId, excludeBrands)) {
+    return null
+  }
+
+  const BrandContainer = ({ children }) => (
```
Comment (klzns) [inline react/components/ProductBrand/index.js]: You can make create this component once, creating this function outside of this render method and passing blockclass.

File: react/components/ProductBrand/index.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+  /** Whether the loading placeholder should have the size of the logo or the text */
+  loadingPlaceholder: PropTypes.oneOf(Object.values(DISPLAY_MODE)),
+  /** Whether it should display the name of the brand if there is no logo */
+  fallbackToText: PropTypes.bool,
+  /** List of brands that should be hidden, if any */
+  excludeBrands: PropTypes.arrayOf(
+    PropTypes.oneOfType([PropTypes.number, PropTypes.string])
+  ),
+  /** Height of the logo */
+  height: PropTypes.nuber,
```
Comment (klzns) [inline react/components/ProductBrand/index.js]: image
```suggestion
  height: PropTypes.number,
```

Comment (lbebber) [reply react/components/ProductBrand/index.js]: 😅

File: react/components/ProductBrand/index.js (hunk 4 above)
Comment (lbebber) [reply react/components/ProductBrand/index.js]: the intention was to specifically avoid passing blockClass every time. but that's fair

File: react/components/ProductBrand/index.js [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+                >
+                  {/** TODO: Use a smarter Image component that handles VTEX image resizing etc. */}
+                  <img
+                    className={generateBlockClass(
+                      styles.productBrandImage,
+                      blockClass
+                    )}
+                    src={changeImageUrlSize(
+                      `/arquivos/ids${imageUrl}`,
+                      // width ? width * dpi : undefined,
```
Comment (jgfidelis) [inline react/components/ProductBrand/index.js]: is this right?

File: react/components/ProductBrand/index.js [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+                      undefined,
+                      height ? height * dpi : undefined
+                    )}
+                    style={{
+                      // width: width || 'auto',
```
Comment (jgfidelis) [inline react/components/ProductBrand/index.js]: commented

File: react/components/ProductBrand/index.js [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+  const BrandContainer = ({ children }) => (
+    <div
+      className={generateBlockClass(styles.productBrandContainer, blockClass)}
+    >
+      {children}
+    </div>
+  )
+
+  const brandNameElement = (
+    <span className={generateBlockClass(styles.productBrandName, blockClass)}>
```
Comment (jgfidelis) [inline react/components/ProductBrand/index.js]: vc nao definiu essa classe no .css

File: react/components/ProductBrand/index.js [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+                <div
+                  style={{
+                    // width,
+                    height,
+                  }}
+                >
+                  {/** TODO: Use a smarter Image component that handles VTEX image resizing etc. */}
+                  <img
+                    className={generateBlockClass(
+                      styles.productBrandImage,
```
Comment (jgfidelis) [inline react/components/ProductBrand/index.js]: nao definiu no .css

File: react/components/ProductBrand/productBrand.css [hunk 10]
```diff
@@ -0,0 +1,3 @@
+.productBrandContainer {}
+.productBrandText {}
```
Comment (jgfidelis) [inline react/components/ProductBrand/productBrand.css]: nao ta sendo usado no js

File: react/components/ProductBrand/productBrand.css [hunk 11]
```diff
@@ -0,0 +1,3 @@
...
+.productBrandLogo {}
```
Comment (jgfidelis) [inline react/components/ProductBrand/productBrand.css]: nao ta sendo usado no js

File: react/components/ProductBrand/index.js (hunk 3 above)
Comment (lbebber) [reply react/components/ProductBrand/index.js]: by crashing and burning

Comment (lbebber) [reply react/components/ProductBrand/index.js]: (updated)

File: react/components/ProductBrand/index.js (hunk 8 above)
Comment (lbebber) [reply react/components/ProductBrand/index.js]: boa

File: react/components/ProductBrand/productBrand.css (hunk 11 above)
Comment (lbebber) [reply react/components/ProductBrand/productBrand.css]: boa 3

File: react/components/ProductBrand/productBrand.css (hunk 10 above)
Comment (lbebber) [reply react/components/ProductBrand/productBrand.css]: boa 2

File: react/components/ProductBrand/index.js (hunk 3 above)
Comment (jgfidelis) [reply react/components/ProductBrand/index.js]: :rage1:

**Pull request merged**